massa-models/src/operation.rs

```diff
@@ -105,6 +152,41 @@ impl OperationId {
             Hash::from_bs58_check(data).map_err(|_| ModelsError::HashError)?,
         ))
     }
+
+    /// split into a tuple of [OperationPrefixId] and [OperationSuffixId]
+    ///
+    pub fn split(&self) -> (OperationPrefixId, OperationSuffixId) {
+        (
+            OperationPrefixId(self.0.to_bytes()[..OPERATION_ID_PREFIX_SIZE_BYTES].to_vec()),
+            OperationSuffixId(self.0.to_bytes()[OPERATION_ID_PREFIX_SIZE_BYTES..].to_vec()),
+        )
+    }
+
+    /// split into a tuple of [OperationPrefixId] and [OperationSuffixId]
+    pub fn into_split(self) -> (OperationPrefixId, OperationSuffixId) {
+        self.split()
+    }
+}
+
+impl OperationPrefixId {
+    /// Retreive an [OperationId] joining the prefix object to his
+    /// corresponding suffix.
+    pub fn join(&self, suffix: &OperationSuffixId) -> ModelsResult<OperationId> {
+        if self.0.len() + suffix.0.len() != OPERATION_ID_SIZE_BYTES {
+            println!(
+                "{}:{}",
+                self.0.len() + suffix.0.len(),
+                OPERATION_ID_SIZE_BYTES
+            );
+            return Err(ModelsError::OperationPrefixJoinError);
+        }
+        let mut id = self.0.clone();
+        id.extend(&suffix.0);
+        // the following code is safe until we correctly check the size in the if
+        // guard and return early an error if the transmutation would fail.
```

Comment:
What do you mean with "until"? Isn't the check made just above? Also we don't seem to return an error if the transmutation would fail?

Reply:
I want to say "because". Technically, I guess that the transmutation here would fail only if the vector has a different size, the value inside have the same size (`u8`). Is it right?